components/script/dom/bindings/utils.rs

```diff
@@ -177,20 +177,20 @@ pub fn get_array_index_from_id(_cx: *mut JSContext, id: HandleId) -> Option<u32>
     }*/
 }
 
-/// Find the index of a string given by `v` in `values`.
+/// Find the enum equivelent of a string given by `v` in `pairs`.
 /// Returns `Err(())` on JSAPI failure (there is a pending exception), and
 /// `Ok((None, value))` if there was no matching string.
-pub unsafe fn find_enum_string_index(cx: *mut JSContext,
-                                     v: HandleValue,
-                                     values: &[&'static str])
-                                     -> Result<(Option<usize>, DOMString), ()> {
+pub unsafe fn find_enum_value<T: Copy>(cx: *mut JSContext,
+                                       v: HandleValue,
+                                       pairs: &[(&'static str, T)])
+                                       -> Result<(Option<T>, DOMString), ()> {
```

Comment:
I think we can avoid the `Copy` bound by doing
```rust
                                       pairs: &'a [(&'static str, T)])
                                       -> Result<(Option<&'a T>, DOMString), ()> {
```

and then in the generated code
```rust
Ok((Some(&value), _)) => value,
```